roxie/ccd/ccdfile.cpp

```diff
@@ -743,7 +743,7 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
             ret->setRemote(true);
         }
         ret->setCache(this);
-        files.setValue(localLocation, (ILazyFileIO *)ret);
+        files.setValue(local->queryFilename(), (ILazyFileIO *)ret);
```

Comment:
Don't understand this change

Reply:
It's because with force remove files on the filename of the IFile created further up is no longer 'locallocation'. And then when Roxie tries to remove it from the 'files' cache by IFile->queryFilename() (see removeCache) - it no longer matches (without the above change).